src/cpp/session/SessionAsyncRProcess.cpp

```diff
@@ -86,51 +91,47 @@ void AsyncRProcess::start(const char* rCommand,
       args.push_back("--internet2");
 #endif
 
-   args.push_back("-e");
+   args.push_back("-f");
    
-   bool needsQuote = false;
-
-   // On Windows, we turn the vector of strings into a single
-   // string to send over the command line, so we must ensure
-   // that the arguments following '-e' are quoted, so that
-   // they are all interpretted as a single argument (rather
-   // than multiple arguments) to '-e'.
-
-#ifdef _WIN32
-   needsQuote = strlen(rCommand) > 0 && rCommand[0] != '"';
-#endif
-
-   std::stringstream command;
-   if (needsQuote)
-      command << "\"";
-
-   std::string escapedCommand = rCommand;
-
-   if (needsQuote)
-      boost::algorithm::replace_all(escapedCommand, "\"", "\\\"");
-    
-   if (rSourceFiles.size())
+   // generate path to temporary file
+   std::string path;
+   error = r::exec::RFunction("base:::tempfile")
+         .addParam("pattern", "rstudio-async-process-")
+         .addParam("tmpdir", workingDir.absolutePathNative())
```

Comment:
I think it'd be preferable to write this temporary file to a temporary folder so there's no chance of it being littered in the case of a crash/etc. I'm also not 100% sure we can guarantee current directory write access in every case where we use AsyncRProcess.